Source/WebKit/Platform/IPC/glib/ConnectionGLib.cpp

```diff
@@ -0,0 +1,408 @@
+                connectionDidClose();
+            }
+            return;
+        }
+
+        if (!bytesRead) {
+            connectionDidClose();
+            return;
+        }
+
+        if (auto decoder = createMessageDecoder(m_readBuffer, m_fileDescriptors))
+            processIncomingMessage(makeUniqueRefFromNonNullUniquePtr(WTFMove(decoder)));
+    }
+}
+
+bool Connection::platformPrepareForOpen()
+{
+    return true;
+}
+
+void Connection::platformOpen()
+{
+    RefPtr<Connection> protectedThis(this);
+    m_isConnected = true;
+
+    m_readSocketMonitor.start(m_socket.get(), G_IO_IN, m_connectionQueue->runLoop(), [protectedThis] (GIOCondition condition) -> gboolean {
+        if (condition & G_IO_HUP || condition & G_IO_ERR || condition & G_IO_NVAL) {
+            protectedThis->connectionDidClose();
+            return G_SOURCE_REMOVE;
+        }
+
+        if (condition & G_IO_IN) {
+            protectedThis->readyReadHandler();
+            return G_SOURCE_CONTINUE;
+        }
+
+        ASSERT_NOT_REACHED();
+        return G_SOURCE_REMOVE;
+    });
+
+    // Schedule a call to readyReadHandler. Data may have arrived before installation of the signal handler.
+    m_connectionQueue->dispatch([protectedThis] {
+        protectedThis->readyReadHandler();
+    });
+}
+
+bool Connection::platformCanSendOutgoingMessages() const
+{
+    return !m_hasPendingOutputMessage;
+}
+
+bool Connection::sendOutgoingMessage(UniqueRef<Encoder>&& encoder)
+{
+    static_assert(sizeof(MessageInfo) + s_attachmentMaxAmount * sizeof(size_t) <= s_messageMaxSize, "Attachments fit to message inline");
+
+    UnixMessage outputMessage(encoder.get());
+    if (outputMessage.attachments().size() > (s_attachmentMaxAmount - 1)) {
+        ASSERT_NOT_REACHED();
+        return false;
+    }
+
+    size_t messageSizeWithBodyInline = sizeof(MessageInfo) + (outputMessage.attachments().size() * sizeof(AttachmentInfo)) + outputMessage.bodySize();
+    if (messageSizeWithBodyInline > s_messageMaxSize && outputMessage.bodySize()) {
+        RefPtr oolMessageBody = WebCore::SharedMemory::allocate(outputMessage.bodySize());
+        if (!oolMessageBody)
+            return false;
+
+        auto handle = oolMessageBody->createHandle(WebCore::SharedMemory::Protection::ReadOnly);
+        if (!handle)
+            return false;
+
+        outputMessage.messageInfo().setBodyOutOfLine();
+
+        memcpySpan(oolMessageBody->mutableSpan(), outputMessage.body());
+
+        outputMessage.appendAttachment(handle->releaseHandle());
+    }
+
+    return sendOutputMessage(WTFMove(outputMessage));
+}
+
+bool Connection::sendOutputMessage(UnixMessage&& outputMessage)
+{
+    ASSERT(!m_hasPendingOutputMessage);
+
+    auto& messageInfo = outputMessage.messageInfo();
+    const auto& attachments = outputMessage.attachments();
+    GOutputVector outputVector[3];
+    int outputVectorLenth = 0;
```

Comment:
typo: 
```suggestion
    int outputVectorLength = 0;
```

Reply:
Carlos either likes Ctrl+V or autocomplete. It's hopeless to expect that typo won't be precisely the same in every single location. :)

Reply:
I love middle click paste :-D